routes/api/project/location/vm.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require "ulid"
+
+class CloverApi
+  hash_branch(:project_location_prefix, "vm") do |r|
+    @serializer = Serializers::Api::Vm
+
+    r.get true do
+      vms = @project.vms_dataset.where(location: @location).authorized(@current_user.id, "Vm:view").all
```

Comment:
`#authorized` is so cool. Perhaps it can be made even less error prone, but it's so powerful.

Reply:
All the time, we give `@current_user.id` as first parameter in `routes`. I couldn't find a way to pass this value in a hidden way.

Hardcoded powers are at everywhere, they feels like open to typos and errors.